data/docs/gcp-monitoring/app-engine/tracing.mdx

```diff
@@ -0,0 +1,398 @@
+```
+
+3. In the service folder, create `server.js` file with the following contents:
+
+```
+const express = require('express');
+const app = express();
+
+app.get('/', (req, res) => {
+  res.send('Hello from App Engine!');
+});
+
+// Listen to the App Engine-specified port, or 8080 otherwise
+const PORT = process.env.PORT || 8080;
+app.listen(PORT, () => {
+  console.log(`Server listening on port ${PORT}...`);
+});
+```
+
+This file contains the actual application logic. Here, we have created a very simple application that opens a GET endpoint at `/` which sends out a statement "Hello from App Engine!" in the response.
+
+4. Next, let us create another file `tracing.js` that contains the instrumentation code to send out traces to SigNoz.
+
+You need to configure the endpoint for SigNoz Cloud in this file. You can find your ingestion key from SigNoz Cloud account details sent on your email.
+
+```
+// tracing.js
+'use strict'
+
+const process = require('process')
+const opentelemetry = require('@opentelemetry/sdk-node')
+const { getNodeAutoInstrumentations } = require('@opentelemetry/auto-instrumentations-node')
+const { OTLPTraceExporter } = require('@opentelemetry/exporter-trace-otlp-http')
+const { Resource } = require('@opentelemetry/resources')
+const { SemanticResourceAttributes } = require('@opentelemetry/semantic-conventions')
+
+// do not set headers in exporterOptions, the OTel spec recommends setting headers through ENV variables
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specifying-headers-via-environment-variables
+
+// highlight-start
+const exporterOptions = {
+  url: 'https://ingest.{region}.signoz.cloud:443/v1/traces',
+}
+// highlight-end
+
+const traceExporter = new OTLPTraceExporter(exporterOptions)
+const sdk = new opentelemetry.NodeSDK({
+  traceExporter,
+  instrumentations: [getNodeAutoInstrumentations()],
+  resource: new Resource({
+    // highlight-next-line
+    [SemanticResourceAttributes.SERVICE_NAME]: 'appengine_nodejs_app',
+  }),
+})
+
+// initialize the SDK and register with the OpenTelemetry API
+// this enables the API to record telemetry
+sdk.start()
+
+// gracefully shut down the SDK on process exit
+process.on('SIGTERM', () => {
+  sdk
+    .shutdown()
+    .then(() => console.log('Tracing terminated'))
+    .catch((error) => console.log('Error terminating tracing', error))
+    .finally(() => process.exit(0))
+})
+```
+
+In this example, we have used `appengine_nodejs_app` as the service name. You can feel free to change it appropriately as per your use-case.
+
+Depending on the choice of your region for SigNoz cloud, the ingest endpoint will vary according to this table.
+
+| Region | Endpoint |
+| --- | --- |
+| US |	ingest.us.signoz.cloud:443 |
+| IN |	ingest.in.signoz.cloud:443 |
+| EU | ingest.eu.signoz.cloud:443 |
+
+5. Modify the `package.json` file with the following contents in the `scripts` and `dependencies` section.
```

Comment:
I will remove the earlier installation steps. I installed it to check it by running locally. These are required to be included in the package.json so that they get packaged.

Reply:
I will add a note instead mentioning the same.

Reply:
Works.